### PR TITLE
Add failsafe for when both data source fetches fail

### PR DESCRIPTION
**Pull request by purple-waves**
In response to one of Sai's comments on the original openAQAdaptor PR

### CODE REVIEW

File: src/main/java/com/google/sps/data/Cache.java [hunk 1]
```diff
@@ -11,23 +11,30 @@
...
-
-    // TODO (Rachel) implement a better failsafe for when both data fetches fail (i.e. a way of checking so the old grid can be returned)
+    int numDataFetchesFailed = 0;
+    
     // Catch the error here because the servlet cannot throw the Exception type
     try {
       data.addAll(getNSWGovData());
     } catch (Exception e) {
       System.out.println(e.getMessage());
+      numDataFetchesFailed ++;
```
Comment (roroross) [inline src/main/java/com/google/sps/data/Cache.java]: but if all the other code in the code base has the style of not having a space before ++, doesnt it make sense to apply that consistency here as well?

Comment (purple-waves) [reply src/main/java/com/google/sps/data/Cache.java]: I couldn't find another example (outside of a for loop) that used this. I'll have another look to double check though

Comment (purple-waves) [reply src/main/java/com/google/sps/data/Cache.java]: None of the java files have it in this case.

Comment (roroross) [reply src/main/java/com/google/sps/data/Cache.java]: but shouldnt it be consistent overall regardless then?

but it is  a nit, and doesnt affect the actual code haha

**Review by roroross — APPROVED** [review]
lgtm

File: src/main/java/com/google/sps/data/Cache.java (hunk 1 above)
Comment (purple-waves) [reply src/main/java/com/google/sps/data/Cache.java]: Ok, I couldn't find any examples in the examples from the starter projects either, but I had it without spaces in that code, so I'll take the spaces out if for nothing else but to try and get myself to write them consistently

Comment (sbland-g) [reply src/main/java/com/google/sps/data/Cache.java]: I couldn't find anything concrete explicitly called out in the style guide; however, I believe it's "commonly accepted" that there should be no space between the variable and post-increment operator.

**Review by sbland-g — APPROVED** [review]

**Pull request merged**